libraries/AdaptiveExpressions/FunctionUtils.cs

```diff
@@ -1072,6 +1105,67 @@ internal static (object, string) NormalizeToDateTime(object timestamp, Func<Date
             return (result, error);
         }
 
+        internal static (CultureInfo, string) TryParseLocale(string localeStr)
+        {
+            CultureInfo result = null;
+            string error = null;
+            try
+            {
+                result = new CultureInfo(localeStr);
+            }
+#pragma warning disable CA1031 // Do not catch general exception types
```

Comment:
Add justification

Reply:
change to the CultureNotFoundException. Also changed the naming of localeStr to locale